crates/typst-pdf/src/color.rs

```diff
@@ -377,26 +378,29 @@ impl PaintEncode for Color {
 /// Extra color space functions.
 pub(super) trait ColorSpaceExt {
     /// Returns the range of the color space.
-    fn range(self) -> [f32; 6];
+    fn range(self) -> &'static [f32];
 
     /// Converts a color to the color space.
-    fn convert<U: QuantizedColor>(self, color: Color) -> [U; 3];
+    fn convert<U: QuantizedColor>(self, color: Color) -> SmallVec<[U; 4]>;
```

Comment:
If the vector is always of length `< 4`, then `ArrayVec` is a bit better than `SmallVec`

Reply:
Ah, you're right! Thanks for that!
I just [changed this line](https://github.com/typst/typst/pull/4568/files#diff-55650bf2dd7b33ec02c37fd3b314eb3a5db0c2aa6bf0baf269ef9c03fe716c95L383-R384) to return an `ArrayVec`.
It seems the `arrayvec` crate was not used, so I've added it [here](https://github.com/typst/typst/pull/4568/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R31).

Reply:
Yeah, that's fine. It's a transitive dependency already anyway.